apps/emqx_bridge_sqlserver/src/emqx_bridge_sqlserver_connector.erl

```diff
@@ -374,9 +376,9 @@ do_get_status(Conn) ->
     case execute(Conn, <<"SELECT 1">>) of
         {selected, [[]], [{1}]} ->
             ok;
-        {error, Reason} ->
-            {error, Reason};
         Other ->
+            %% force restart if SELECT 1 returns error
+            disconnect(Conn),
```

Comment:
Not needed if we use `auto_reconnect` and return `?status_disconnected` on errors.

Reply:
not sure I understand

Reply:
without this disconnect here, the added test does not pass [here](https://github.com/emqx/emqx/pull/14897/files#diff-9460ba192c8e462aab2e4c59631f37c2fdbabe170e47af19e9447c986cc7cec7R321)

if returning `disconnected` status from health-check would trigger the entire pool to restart,
it does not explain why the odbc connection was not restarted (the on_stop did not happen?)
I thought it was because this connector does not call emqx_resource:allocate_resource, but after a closer look that does not seem to be related.